upup/pkg/fi/cloudup/urls.go

```diff
@@ -118,39 +119,44 @@ func SetKopsAssetsLocations(assetsBuilder *assets.AssetBuilder) error {
 }
 
 // NodeUpAsset returns the asset for where nodeup should be downloaded
-func NodeUpAsset(assetsBuilder *assets.AssetBuilder) (*MirroredAsset, error) {
-	// Avoid repeated logging
-	if nodeUpAsset != nil {
+func NodeUpAsset(assetsBuilder *assets.AssetBuilder, arch architectures.Architecture) (*MirroredAsset, error) {
+	if nodeUpAsset == nil {
+		nodeUpAsset = make(map[architectures.Architecture]*MirroredAsset)
+	} else if nodeUpAsset[arch] != nil {
 		// Avoid repeated logging
-		klog.V(8).Infof("Using cached nodeup location: %v", nodeUpAsset.Locations)
-		return nodeUpAsset, nil
+		klog.V(8).Infof("Using cached nodeup location: %v", nodeUpAsset[arch].Locations)
```

Comment:
Include arch in log?

Reply:
I thought that most locations already have the arch in the name.

Reply:
Maybe, maybe not. Could be a bug where the arch in the location is wrong.

Reply:
OK, done.